### PR TITLE
Fixes #7185: Stash and unstash changes when changing current branch with rudder-dev

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/7185

### CODE REVIEW

**Comment by peckpeck** [comment]
PR updated

**Comment by peckpeck** [comment]
PR updated

**Comment by peckpeck** [comment]
PR updated

**Comment by peckpeck** [comment]
PR updated

**Comment by peckpeck** [comment]
PR updated

**Comment by peckpeck** [comment]
PR updated

**Comment by peckpeck** [comment]
PR updated

**Comment by peckpeck** [comment]
PR updated

File: scripts/rudder-dev/rudder-dev [hunk 1]
```diff
@@ -1177,7 +1234,7 @@ if __name__ == "__main__":
     if re.match(r'^i?\d{2,5}$', smart):
       arguments['branch'] = True
       arguments['<ticket_id>'] = smart
-    elif re.match(r'^\d\.\d+$|^master$', smart):
+    elif re.match(r'^\d\.(\d+|x)$|^master$', smart):
```
Comment (VinceMacBuche) [inline scripts/rudder-dev/rudder-dev]: The line could start by v no ? (like v0.x ?)

Comment (peckpeck) [inline scripts/rudder-dev/rudder-dev]: This is the version name, not the branch name

**Comment by peckpeck** [comment]
PR updated

**Comment by peckpeck** [comment]
PR updated

**Pull request merged**